src/vs/editor/browser/widget/diffEditorWidget2/diffModel.ts

```diff
@@ -171,6 +182,9 @@ export class DiffModel extends Disposable implements IDiffEditorViewModel {
 	}
 
 	public ensureModifiedLineIsVisible(lineNumber: number, tx: ITransaction): void {
+		if (this.diff.get()?.mappings.length === 0) {
```

Comment:
should we also short circuit if `this.diff.get()` is undefined? Just noticed that you're using `?`

Reply:
When diff is undefined, there are no ranges so it does not matter. But I agree, this code needs some clean up once the feature is stabilized.